BGL/include/CGAL/boost/graph/Euler_operations.h

```diff
@@ -585,6 +585,12 @@ add_face(const VertexRange& vr, Graph& g)
 
   std::vector<vertex_descriptor> vertices(vr.begin(), vr.end()); // quick and dirty copy
   unsigned int n = (unsigned int)vertices.size();
+  //check that every vertex is unique
+  std::sort(vertices.begin(), vertices.end());
+  if(std::unique(vertices.begin(), vertices.end()) != vertices.end())
```

Comment:
Actually, [`std::adjacent_find`] is better than [`std::unique`], because it stops as soon as it has found two adjacent identical values:
```suggestion
  if(std::adjacent_find(vertices.begin(), vertices.end()) != vertices.end())
```

[`std::adjacent_find`]: https://en.cppreference.com/w/cpp/algorithm/adjacent_find
[`std::unique`]: https://en.cppreference.com/w/cpp/algorithm/unique